src/controllers/api/products.js

```diff
@@ -0,0 +1,259 @@
+
+        // IF product DOES NOT EXIST
+        if (!product) {
+            return res.status(404).json({
+                success: false,
+                message: `Product with ID: ${id}, does not exist in database`,
+            });
+        } else {
+            // IF product DOES EXIST
+            return res.status(200).json({
+                success: true,
+                data: product,
+            });
+        }
+    } catch (err) {
+        return res.status(500).json({
+            success: false,
+            error: err.message,
+        });
+    }
+};
+
+const createProduct = async(req, res) => {
+    try {
+        const { productName, price, stock, tagIds } = req.body;
+
+        // check if valid request body
+        if (!productName || !price || !stock) {
+            return res.status(422).json({
+                success: false,
+                message: "Oops!! A value within the request body was not valid",
+            });
+        } else {
+            // find product by product_name
+            const productExists = await Product.findOne({
+                where: {
+                    productName: productName,
+                },
+            });
+
+            // IF product EXISTS
+            if (productExists) {
+                return res.status(404).json({
+                    success: false,
+                    message: `Oops!! ${productName} already exists in database`,
+                });
+            } else {
+                // create a new Product
+                const newProduct = await Product.create({
+                    productName: capitalizeString(productName),
+                    price,
+                    stock,
+                }, {
+                    include: {
+                        model: Category,
+                    },
+                });
+
+                // check IF updated product has associated TAGS to create
+                if (tagIds.length) {
+                    // create an object with the product ID related to each tag provided
+                    const productTagIdArr = tagIds.map((tagId) => {
+                        return {
+                            productId: newProduct.id,
+                            tagId,
+                        };
+                    });
+                    // create a new ProductTag using productTagIdArr(ProductIDs and TagIDs)
+                    await ProductTag.bulkCreate(productTagIdArr);
+                }
+
+                // return if product was created successfully
+                return res.status(200).json({
+                    success: true,
+                    productData: newProduct,
+                    tagIds,
+                });
+            }
+        }
+    } catch (err) {
+        return res.status(500).json(err.message);
+    }
+};
+
+const updateProductById = async(req, res) => {
+    try {
+        const { id } = req.params;
+        const { productName, price, stock, tagIds } = req.body;
```

Comment:
You do not extract the categoryId from the req.body here. This means that user is unable to update the categoryId.